### PR TITLE
chore: Update create app template to align with example app

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Some minor maintenance 🧹 to the `create-app` plugin to ensure the resulting app better aligns to the example in the template repo, including,

* Moving test-utils to a devDependency
* Minor documentation tweaks
* Bump of `@spofity/prettier-config` version

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by adamdmharvey** [comment]
An interesting note on this. Discovered the create-app does not import the `@backstage/search-common` package - but seems it doesn't need it yet?  Should this be included in some way? I checked the PR where it was added, and it was added to the example app but not the templated one yet. (and again seems to work without it)

cc @realandersn may know

File: .changeset/thirty-readers-attack.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/create-app': patch
+---
+
+Alignments of template with Backstage example app in monorepo
```
Comment (Rugvip) [inline .changeset/thirty-readers-attack.md:5]: I think it's alright to not describe the reordering and readme changes here, but I'd say it's best if we mention the `@spotify/prettier-config` version bump

File: packages/create-app/README.md [hunk 2]
```diff
@@ -22,4 +22,4 @@ yarn backstage-create-app
 ## Documentation
 
 - [Backstage Readme](https://github.com/backstage/backstage/blob/master/README.md)
-- [Backstage Documentation](https://github.com/backstage/backstage/blob/master/docs/README.md)
+- [Backstage Documentation](hhttps://backstage.io/docs/)
```
Comment (Rugvip) [inline packages/create-app/README.md:25]:
```suggestion
- [Backstage Documentation](https://backstage.io/docs/)
```

**Review by Rugvip — COMMENTED** [review]
Nice, thanks! 😁 🧹 ✨

**Comment by adamdmharvey** [comment]
There's a few more things about this. For e.g., there's a react hot loading call missing from the template, and a few others, but I'll call it here with just a trivial enhancement.

**Review by jhaals — APPROVED** [review]
Thank you Adam! ⭐

**Pull request merged**